lib/Backends/BackendUtils.cpp

```diff
@@ -81,6 +74,55 @@ runtime::RuntimeBundle::getSymbolInfo(const Named *v) const {
   return it->second;
 }
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH) {
+  // Check if trainable, if it is return true.
+  if (PH->isTraining()) {
```

Comment:
Why is this correct? The isTraining flag is only used by the differentiation method, not by the backend.

Reply:
The thought was if a PH is marked as trainable, it would always be an input and output, but that assumption only holds for training, not inference. Check removed.